### PR TITLE
refactor(callout): new hideTitle and hideBody props and a component slot

### PR DESCRIPTION
**Pull request by masoudmanson**
## Summary

**Callout**
Github issue: #909 

Refactor the `Callout` component with the following updates:

- Add a component slot for `extraContent`, allowing additional content to be placed at the bottom of the `Callout` component.
- Introduce two new props to toggle the visibility of the `calloutTitle` and `calloutText`.

## Checklist

- [ ] Default Story in Storybook
- [ ] LivePreview Story in Storybook
- [ ] Test Story in Storybook
- [ ] Tests written
- [ ] Variables from `defaultTheme.ts` used wherever possible
- [ ] If updating an existing component, depreciate flag has been used where necessary
- [ ] Chromatic build verified by @chanzuckerberg/sds-design

### CODE REVIEW

**Review by clarsen-czi — CHANGES_REQUESTED** [review]
Looks great @masoudmanson! Once you get the alignment of the title with the icon when there's no body text updated it will be good to go

**Comment by masoudmanson** [comment]
> Looks great @masoudmanson! Once you get the alignment of the title with the icon when there's no body text updated it will be good to go

Thanks so much, @clarsen-czi! I’ve updated the title styles. Let me know if they need further refactoring or if we’re good to go! 🥳🎉🙏🏻

**Review by clarsen-czi — APPROVED** [review]
Looks great, @masoudmanson! Thanks for the fix!

**Review by tihuan — APPROVED** [review]
Looks fantastic! Thanks so much for the upgrade, @masoudmanson 🤩 🙏 

Just one small non blocking discussion 😁 Thanks again!

File: packages/components/src/core/Callout/components/CalloutExtraContent/style.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  hideBody?: boolean;
+}
+
+export const StyledCalloutExtraContent = styled("div")`
+  ${(props: CommonThemeProps & CalloutExtraContentProps) => {
+    const { hideTitle = false, hideBody = false } = props;
+    const spaces = getSpaces(props);
+
+    return `
+      margin: ${!hideTitle || !hideBody ? spaces?.m : 0}px 0 0 0;
```
[the remark this answers is not in the document]
Comment (masoudmanson) [reply packages/components/src/core/Callout/components/CalloutExtraContent/style.ts:15]: Ahh, you’re absolutely right 🤣 My code does look funny 😱🤣 Thanks for catching that, @tihuan! 
Always learning from you 🥳😍

[the remark this answers is not in the document]
Comment (tihuan) [reply packages/components/src/core/Callout/components/CalloutExtraContent/style.ts]: You too!! Thanks again for the great work 😄 👏 🥇 ❤️

**Pull request merged**